dropshot/tests/test_config.rs

```diff
@@ -542,6 +544,14 @@ async fn test_config_handler_task_mode_cancel() {
 
     server.close().await.unwrap();
 
+    // Wait for the log to finish.
+    sleep(Duration::from_millis(100)).await;
```

Comment:
This is not great -- it's going to lead to flaky tests.  Elsewhere we use [this function](https://github.com/oxidecomputer/omicron/blob/2efb6148774bd56668e9322e180d9cd793c8d620/test-utils/src/dev/poll.rs#L32-L91) to keep checking a condition until some maximum time has elapsed.  That's a lot more robust.  But it's not easy to use here.  You could duplicate the pattern here (it's basically just a loop until some maximum timeout has elapsed).  I'm not sure it's worth testing this though -- I don't think we have other tests that verify log entries.  (Not that it would be a bad idea!)

Reply:
I agree, this is pretty janky. I hadn't seen the `wait_for_condition` function although it's probably not worth starting to verify log entries in this PR so I've removed that whole bit.